lib/agenda.js

```diff
@@ -427,17 +427,16 @@ Agenda.prototype._findAndLockNextJob = function(jobName, definition, cb) {
   if (s.topology.connections().length === 0) {
     cb(new Error( 'No MongoDB Connection'));
   } else {
-    this._collection.findAndModify(
+    this._collection.findOneAndUpdate(
       {
         $or: [
           {name: jobName, lockedAt: null, nextRunAt: {$lte: this._nextScanAt}, disabled: { $ne: true }},
           {name: jobName, lockedAt: {$exists: false}, nextRunAt: {$lte: this._nextScanAt}, disabled: { $ne: true }},
           {name: jobName, lockedAt: {$lte: lockDeadline}, disabled: { $ne: true }}
         ]
       },
-      {'priority': -1},  // sort
       {$set: {lockedAt: now}},  // Doc
-      {'new': true},  // options
+      {returnOriginal: false, 'priority': -1},  // options & sort
```

Comment:
I've just noticed that according to the [findOneAndUpdate docs](http://mongodb.github.io/node-mongodb-native/2.2/api/Collection.html#findOneAndUpdate) this is incorrect. It should be:

```js
{returnOriginal: false, sort: {priority: -1}}
```

There doesn't seem to be any test coverage for this either. I'll put in another PR to fix.